src/libraries/Microsoft.Extensions.Http/src/HttpClientFactoryOptions.cs

```diff
@@ -16,15 +17,20 @@ public class HttpClientFactoryOptions
     {
         // Establishing a minimum lifetime helps us avoid some possible destructive cases.
         //
-        // IMPORTANT: This is used in a resource string. Update the resource if this changes.
+        // IMPORTANT: MinimumHandlerLifetime is used in a resource string. Update the resource if this changes.
         internal static readonly TimeSpan MinimumHandlerLifetime = TimeSpan.FromSeconds(1);
 
         private TimeSpan _handlerLifetime = TimeSpan.FromMinutes(2);
 
         /// <summary>
         /// Gets a list of operations used to configure an <see cref="HttpMessageHandlerBuilder"/>.
         /// </summary>
-        public IList<Action<HttpMessageHandlerBuilder>> HttpMessageHandlerBuilderActions { get; } = new List<Action<HttpMessageHandlerBuilder>>();
+        /// <remarks>
+        /// Accessing this property will have a performance impact, as it will disable some optimizations.
+        /// Consider using configuration APIs like ConfigurePrimaryHttpMessageHandler or
+        /// ConfigureAdditionalHttpMessageHandlers instead of accessing this property directly.
```

Comment:
Will this regress existing code, or are the optimizations this disables only relevant for code using new features?